lib/Widgets/SettingsPage/AbstractSimpleSettingsPage.vala

```diff
@@ -0,0 +1,159 @@
+     */
+    public Gtk.Switch status_switch;
+
+    /**
+     * Creates a #Gtk.Switch #status_switch in the header of #this
+     */
+    public bool activatable { get; construct; }
+
+    /**
+     * Creates a #Gtk.Label with a page description in the header of #this
+     */
+    public string description {
+        get {
+            return _description;
+        } 
+        construct set {
+            if (description_label != null) {
+                description_label.label = value;
+            }
+            _description = value;
+        }
+    }
+
+    /**
+     * An icon name associated with #this
+     */
+    public string icon_name {
+        get {
+            return _icon_name;
+        } 
+        construct set {
+            if (header_icon != null) {
+                header_icon.icon_name = value;
+            }
+            _icon_name = value;
+        }
+    }
+
+    /**
+     * A title associated with #this
+     */
+    public string title {
+        get {
+            return _title;
+        } 
+        construct set {
+            if (title_label != null) {
+                title_label.label = value;
+            }
+            _title = value;
+        }
+    }
+
+    /**
+     * Creates a new SimpleSettingsPage
+     */
+    public SimpleSettingsPage () {
+        Object (activatable: activatable,
+                icon_name: icon_name,
+                description: description,
+                title: title);
+    }
+
+    construct {
+        header_icon = new Gtk.Image.from_icon_name (icon_name, Gtk.IconSize.DIALOG);
+        header_icon.pixel_size = 48;
+        header_icon.valign = Gtk.Align.START;
+
+        title_label = new Gtk.Label (title);
+        title_label.xalign = 0;
+        title_label.get_style_context ().add_class ("h2");
+
+        var header_area = new Gtk.Grid ();
+        header_area.column_spacing = 12;
+        header_area.row_spacing = 3;
+        header_area.attach (header_icon, 0, 0, 1, 2);
+        header_area.attach (title_label, 1, 0, 1, 1);
+
+        if (description != null) {
+            description_label = new Gtk.Label (description);
+            description_label.xalign = 0;
+            description_label.wrap = true;
+
+            header_area.attach (description_label, 1, 1, 1, 1);
+        }
+
+        if (activatable) {
+            status_switch = new Gtk.Switch ();
+            status_switch.hexpand = true;
+            status_switch.halign = Gtk.Align.END;
+            status_switch.valign = Gtk.Align.CENTER;
+            header_area.attach (status_switch, 2, 0, 1, 1);
+        }
+
+        content_area = new Gtk.Grid ();
+        content_area.column_spacing = 12;
+        content_area.row_spacing = 12;
+        content_area.vexpand = true;
+
+        action_area = new Gtk.ButtonBox (Gtk.Orientation.HORIZONTAL);
+        action_area.set_layout (Gtk.ButtonBoxStyle.END);
+        action_area.set_spacing (6);
```

Comment:
Could we use `=` as we use it everywhere in this class?
`action_area.spacing = 6`